### PR TITLE
Improve Channel doc snippets

### PR DESCRIPTION
**Pull request by zsmb13**
### Description

Lots of readability / style improvements for Channel doc snippets, including the completion of https://stream-io.atlassian.net/browse/CAS-645

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] ~Changelog updated with client-facing changes~
- [ ] ~New code is covered by unit tests~
- [ ] ~Comparison screenshots added for visual changes~
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java [hunk 1]
```diff
@@ -39,11 +39,22 @@
      */
     class ChannelInitialization {
         public void initialization() {
-            // Create channel client using channel type and channel id
-            ChannelClient channelClient = client.channel("channel-type", "channel-id");
+            ChannelClient channelClient = client.channel("messaging", "channel-id");
```
Comment (adasiewiczr) [inline stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java:42]:
```suggestion
            ChannelClient channelClient = client.channel("messaging", "general");
```

File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java [hunk 2]
```diff
@@ -52,67 +63,149 @@ public void initialization() {
...
+            );
+            int offset = 0;
+            int limit = 10;
+            QuerySort<Channel> sort = new QuerySort<Channel>().desc("last_message_at");
+            int messageLimit = 0;
+            int memberLimit = 0;
+
+            QueryChannelsRequest request = new QueryChannelsRequest(filter, offset, limit, sort, messageLimit, memberLimit)
+                    .withWatch() // Watches the channels automatically
+                    .withState();
```
Comment (adasiewiczr) [inline stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java:109]: Maybe we should also explain what `withState()` does?

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/Channels.kt [hunk 3]
```diff
@@ -28,11 +31,22 @@ class Channels(val client: ChatClient, val channelClient: ChannelClient) {
      */
     inner class ChannelInitialization {
         fun initialization() {
-            // Create channel client using channel type and channel id
-            val channelClient = client.channel("channel-type", "channel-id")
-
-            // Or create channel client using channel cid
-            val anotherChannelClient = client.channel("cid")
+            val channelClient = client.channel("messaging", "channel-id")
```
Comment (adasiewiczr) [inline stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/Channels.kt:34]:
```suggestion
            val channelClient = client.channel(channelType = "messaging", channelId = "general")
```

File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java (hunk 2 above)
Comment (zsmb13) [reply stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Channels.java:109]: Explained in the written part of the docs to some degree, but will double-check its behaviour later

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**